src/traits/states.rs

```diff
@@ -12,14 +12,14 @@ use crate::{
         BlockPayload,
     },
 };
-use serde::{de::DeserializeOwned, Serialize};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
 use std::{error::Error, fmt::Debug, future::Future, hash::Hash};
 
 /// Instance-level state, which allows us to fetch missing validated state.
 pub trait InstanceState: Debug + Send + Sync {}
 
 /// Application-specific state delta, which will be used to store a list of merkle tree entries.
-pub trait StateDelta: Debug + Send + Sync {}
+pub trait StateDelta: Debug + Send + Sync + Serialize + for<'a> Deserialize<'a> {}
```

Comment:
Can you run this by the sequencer team?  This might be a breaking change for them

Reply:
We don't implement this in the sequencer yet. Anyways, this should be fine because all merkle tree-related structures implement Serialize/Deserialize.

Reply:
It shouldn't matter to VBS.